core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -363,6 +344,10 @@ public RetryOnReplicaException(StreamInput in) throws IOException {
     }
 
     private final class AsyncReplicaAction extends AbstractRunnable {
+        /**
+         * The number of operations remaining before we can reply. See javadoc for {@link #operationComplete()} more.
+         */
+        private final AtomicInteger operationsUntilReply = new AtomicInteger(2);
```

Comment:
Actually I've figured out another way to remove it that is more like the way @bleskes removed the stash for the async on the primary so I'm going to give that a shot.